### PR TITLE
nixos/tests/forgejo: refactor and test Forgejo Actions workflow, nixos/gitea-actions-runner: set the `$HOME` environment variable

### PR DESCRIPTION
**Pull request by emilylange**
## Things done

Please read the commit messages for details :)

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cole-h** [comment]
You'll want to rebase on the latest `master` -- there was an issue that broke ofborg's ability to evaluate Nixpkgs that has since been resolved. Sorry for the inconvenience!

**Review by adamcstephens — COMMENTED** [review]
Nice work, this is cool. :)

File: nixos/modules/services/continuous-integration/gitea-actions-runner.nix [hunk 1]
```diff
@@ -203,6 +203,8 @@ in
             TOKEN = "${instance.token}";
           } // optionalAttrs (wantsPodman) {
             DOCKER_HOST = "unix:///run/podman/podman.sock";
+          } // {
+            HOME = "/var/lib/gitea-runner/${name}";
```
Comment (adamcstephens) [inline nixos/modules/services/continuous-integration/gitea-actions-runner.nix:207]: Are there negative implications for this change? For example will a local runner create files in this directory?

Comment (emilylange) [reply nixos/modules/services/continuous-integration/gitea-actions-runner.nix:207]: On my `:host` runner I see an additional `.cache` directory, which holds `act` and `actcache`.

`.cache/act/actions-checkout@v4` for example.

We could probably use `/var/cache/` instead /shrug

I can't think of any negative implications either way though.

Comment (adamcstephens) [reply nixos/modules/services/continuous-integration/gitea-actions-runner.nix:207]: Yeah, I'm not too worries about the act directories as long as other state isn't getting stored there. Adding `CacheDirectory` to the serviceConfig and pointing to it seems reasonable, but isn't a huge deal.

**Review by adamcstephens — APPROVED** [review]

**Review by yayayayaka — APPROVED** [review]

**Review by bendlas — APPROVED** [review]

File: nixos/modules/services/continuous-integration/gitea-actions-runner.nix (hunk 1 above)
Comment (Skarlett) [reply nixos/modules/services/continuous-integration/gitea-actions-runner.nix:207]: Without the addition of `HOME`, my forgejo-runner breaks, with particularly hard to debug error messages.


when using
`echo "all good"`
```
bash: .cache/act/1a25cf138c689101/act/workflow/0.sh: No such file or directory
```

When using `actions/checkout`
```
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] setupEnv => map[ACT:true ACTIONS_CACHE_URL:http://10.0.0.5:43687/ ACTIONS_RESULTS_URL:http://codex.cypress.local ACTIONS_RU>
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] evaluating expression ''
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] expression '' evaluated to 'false'
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] Skipping step 'cat testfile' due to ''
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] setupEnv => map[ACT:true ACTIONS_CACHE_URL:http://10.0.0.5:43687/ ACTIONS_RESULTS_URL:http://codex.cypress.local ACTIONS_RU>
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] evaluating expression 'always()'
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] expression 'always()' evaluated to 'true'
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] ⭐ Run Post http://codex.cypress.local/actions/checkout@main
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] run post step for 'http://codex.cypress.local/actions/checkout@main'
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] executing remote job container: [node .cache/act/28cb059d3c646cfe/act/actions/http---codex.cypress.local-actions-checkout@m>
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   | node:internal/modules/cjs/loader:1137
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |   throw err;
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |   ^
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   | Error: Cannot find module '/var/lib/private/gitea-runner/host/.cache/act/28cb059d3c646cfe/hostexecutor/.cache/act/28cb059d3c646>
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |     at Module._resolveFilename (node:internal/modules/cjs/loader:1134:15)
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |     at Module._load (node:internal/modules/cjs/loader:975:27)
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:128:12)
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |     at node:internal/main/run_main_module:28:49 {
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |   code: 'MODULE_NOT_FOUND',
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |   requireStack: []
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   | }
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   | Node.js v18.19.1
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   ❌  Failure - Post http://codex.cypress.local/actions/checkout@main
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] Cleaning up container for job checkout-cat
```

If you're running this on 23.11
```
systemd.services.gitea-runner-<instance>.environment.HOME = "/var/lib/gitea-runner/<instance>";
```
will clear up any strange errors related to this issue.

**Review by Skarlett — COMMENTED** [review]
Tested and working.

**Pull request merged**